### PR TITLE
Add DTH test for EnumerateProjectContexts

### PR DESCRIPTION
**Pull request by troydai**
/cc @JunTaoLuo @davidfowl @victorhurdugaci

### CODE REVIEW

File: test/Microsoft.Dnx.Testing.Framework/DnxSdk/Dth/DthMessageTypes.cs [hunk 1]
```diff
@@ -9,8 +9,10 @@ public class DthMessageTypes
         public const string GetDiagnostics = nameof(GetDiagnostics);
         public const string ProtocolVersion = nameof(ProtocolVersion);
         public const string ChangeConfiguration = nameof(ChangeConfiguration);
+        public const string EnumerateProjectContexts = nameof(EnumerateProjectContexts);
 
         // responses
         public const string AllDiagnostics = nameof(AllDiagnostics);
+        public const string ProjectContexts = nameof(ProjectContexts);
```
Comment (victorhurdugaci) [inline test/Microsoft.Dnx.Testing.Framework/DnxSdk/Dth/DthMessageTypes.cs]: Did you just assign a constant to a constant just so you don't use the other constant? :laughing:

File: test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs [hunk 2]
```diff
@@ -437,6 +437,35 @@ public void DthCompilation_ChangeConfiguration(DnxSdk sdk)
...
+                var projects = new Project[]
+                {
+                    solution.GetProject("EmptyLibrary"),
+                    solution.GetProject("UnresolvedPackageSample"),
+                    solution.GetProject("EmptyConsoleApp")
+                };
+
+                var contexts = projects.ToDictionary(proj => proj.ProjectDirectory,
+                                                     proj => client.Initialize(proj.ProjectDirectory));
+                client.DrainMessage(21);
```
Comment (victorhurdugaci) [inline test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs]: What's 21?

Comment (JunTaoLuo) [inline test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs]: Limits the number of messages to be collected/drained. This speeds up the tests.

Comment (troydai) [inline test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs]: Exactly. 21 = 7 * 3, we got seven response for each project initialization.

21 => 2+1 => 3 Half Life 3!

**Comment by troydai** [comment]
Any more? Sign off? @victorhurdugaci @JunTaoLuo

File: test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs (hunk 2 above)
Comment (victorhurdugaci) [inline test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs]: Add a comment explaining that

**Comment by victorhurdugaci** [comment]
:shipit: after adding the comment

**Pull request merged**